src/types/index.ts

```diff
@@ -0,0 +1,10 @@
+import type { employmentTypes } from "@/sample-data";
+
+export type JobExperienceType = {
+  title: string;
+  employmentType: (typeof employmentTypes)[number];
+  company: string;
+  isCurrent: boolean;
+  startDate: Date;
+  endDate?: Date;
+};
```

Comment:
This type is fine, but allows for inconsistent data - e.g. a job with `isCurrent` true, that has both startDate and endDate set.
To better reflect the structure of the data, you could use a union:

```
type Job = {
  job_title: string;
  employment_type: string;
  company: string;
  start_date: { month: string; year: string };
};

type CurrentJob = Job & {
  is_current: true;
};

type PreviousJob = Job & {
  is_current: false;
  end_date: { month: string; year: string };
};

export type JobExperience = CurrentJob | PreviousJob;
```

See https://reactpractice.dev/solution/tutorial-build-the-linkedin-add-experience-form/